src/network/network_client.h

```diff
@@ -112,9 +112,15 @@ typedef ClientNetworkGameSocketHandler MyClient;
 void NetworkClient_Connected();
 void NetworkClientSetCompanyPassword(const char *password);
 
-extern CompanyID _network_join_as;
+/** Information required to join a server. */
+struct NetworkJoinInfo {
+	NetworkJoinInfo() : company(COMPANY_SPECTATOR), server_password(nullptr), company_password(nullptr) {}
+	NetworkAddress address;          ///< The address of the server to join.
+	CompanyID      company;          ///< The company to join.
+	const char    *server_password;  ///< The password of the server to join.
```

Comment:
any chance we can make this into std::strings while we are here? ("no" is a valid answer)

Reply:
Not in this at this moment, as then I'd also need to put parts of other PRs into here like sending std::string_view into a packet.